typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,42 @@
+const $tabs = document.querySelectorAll('.tab');
+if (!$tabs) throw new Error('.tab query failed!');
+
+const $tabContainer = document.querySelector('.tab-container');
+if (!$tabContainer) throw new Error('.tab-container query failed');
+
+const $views = document.querySelectorAll('.view');
+if (!$views) throw new Error('view query failed!');
+
+let currentTabNdx: number = 0;
+
+$tabContainer.addEventListener('click', (event: Event) => {
+  const eventTarget = event.target as HTMLElement;
+  console.log('eventTarget:', eventTarget);
+
+  if (!(eventTarget.className === 'tab')) {
+    console.log('not an inactive tab!');
+    return;
+  } // if what we clicked on was not a tab, return
```

Comment:
What happens if you click on an active tab though? There's a method we talk about that would be better used here.

Reply:
if we click on an active tab then shouldn't the function just return like it does here since we don't want to execute anything?

Reply:
That is true. Since there isn't much of a specific case where it would matter here I'll approve. There could be something in the future if you were working with a project that pulls data from elsewhere that you may want to refresh if you tab was clicked again 🤔